source/velocity_boundary_conditions/ascii_data.cc

```diff
@@ -0,0 +1,439 @@
+                             "in the input file, this number is either interpreted as seconds or as years.");
+          prm.declare_entry ("First data file number", "0",
+                             Patterns::Integer (),
+                             "Number of the first velocity file to be loaded when the model time "
+                             "is larger than 'First velocity file model time'.");
+          prm.declare_entry ("Decreasing file order", "false",
+                             Patterns::Bool (),
+                             "In some cases the boundary files are not numbered in increasing "
+                             "but in decreasing order (e.g. 'Ma BP'). If this flag is set to "
+                             "'True' the plugin will first load the file with the number "
+                             "'First velocity file number' and decrease the file number during "
+                             "the model run.");
+          prm.declare_entry ("Scale factor", "1",
+                             Patterns::Double (0),
+                             "Scalar factor, which is applied to the boundary velocity. "
+                             "You might want to use this to scale the velocities to a "
+                             "reference model (e.g. with free-slip boundary) or another "
+                             "plate reconstruction. Another way to use this factor is to "
+                             "convert units of the input files. The unit is assumed to be"
+                             "m/s or m/yr depending on the 'Use years in output instead of "
+                             "seconds' flag. If you provide velocities in cm/yr set this "
+                             "factor to 0.01.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    AsciiData<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Boundary velocity model");
+      {
+        prm.enter_subsection("Ascii data model");
+        {
+          // Get the path to the data files. If it contains a reference
+          // to $ASPECT_SOURCE_DIR, replace it by what CMake has given us
+          // as a #define
+          data_directory    = prm.get ("Data directory");
+          {
+            const std::string      subst_text = "$ASPECT_SOURCE_DIR";
+            std::string::size_type position;
+            while (position = data_directory.find (subst_text),  position!=std::string::npos)
+              data_directory.replace (data_directory.begin()+position,
+                                      data_directory.begin()+position+subst_text.size(),
+                                      ASPECT_SOURCE_DIR);
+          }
+
+          data_file_name    = prm.get ("Data file name");
+
+          scale_factor      = prm.get_double ("Scale factor");
+          data_points[0]    = prm.get_double ("Number of x grid points");
```

Comment:
For now I am a bit hesitant to request from every data file to have the POINTS: keyword in it, in particular since the grid is not allowed to change over time anyway. That is why the implementation currently only enforces to get the number of points either from the input file, or from the first data file, or from all data files. If you set the points in the input file, and in the data files, then an assertion ensures that there is no contradiction. If you strongly feel for removing the input parameter and enforce to have it in at least the first data file (or all data files?), then we can remove it.